proposals/vector_to_regular_points.json

```diff
@@ -23,12 +23,22 @@
         },
         {
             "name": "distance",
-            "description": "Defines the minimum distance in the unit of the reference system that is required between two samples generated *inside* a single geometry.\n\n- For **polygons**, the distance defines the cell sizes of a regular grid that starts at the upper-left bound of each polygon. The centroid of each cell is then a sample point. If the centroid is not enclosed in the polygon, no point is sampled. If no point can be sampled for the geometry at all, the first coordinate of the geometry is returned as point.\n- For **lines** (line strings), the sampling starts with a point at the first coordinate of the line and then walks along the line and samples a new point each time the distance to the previous point has been reached again.\n- For **points**, the point is returned as given.",
+            "description": "Defines the minimum distance in meters that is required between two samples generated *inside* a single geometry. If the unit of the spatial reference system is not meters, a `UnitMismatch` error is thrown. Use ``vector_reproject()`` to convert to a suitable spatial reference system.\n\n- For **polygons**, the distance defines the cell sizes of a regular grid that starts at the upper-left bound of each polygon. The centroid of each cell is then a sample point. If the centroid is not enclosed in the polygon, no point is sampled. If no point can be sampled for the geometry at all, the first coordinate of the geometry is returned as point.\n- For **lines** (line strings), the sampling starts with a point at the first coordinate of the line and then walks along the line and samples a new point each time the distance to the previous point has been reached again.\n- For **points**, the point is returned as given.",
             "schema": {
                 "type": "number",
                 "minimumExclusive": 0
             }
         },
+        {
+            "name": "unit",
+            "description": "Specifies the unit the distance is given in. The only allowed value is meters (`m`), which is the default value. This parameter exists to be future-proof and might be extended in the future.",
```

Comment:
any reason this parameter wasn't added  to `vector_buffer`?

I'm also not sure it's necessary to define this parameter to be future-proof. Just omitting it now and having meter by default later will also be future proof I think

Reply:
Yes, we can simply add it there as it's the last parameter in the list.
We need to add it in vector_to_regular_points.json because it's not the last parameter and there are languages like JS where the order or params matters.

Reply:
>  Just omitting it now and having meter by default later will also be future proof I think

No, not all languages allow Parameters by name like Python.

Reply:
ok I understand.

If you want to keep distance and unit better together, isn't it more logical to use group these aspects in a single object, like we currently already do in apply_neightborhood? This would allow to omit it now and add support for a "distance object" later in a future-proof way.

Reply:
>  (instead of creating wrapper objects that - right now - would look very meaningless).

well, I meant to just define `distance` as a simple number for now (implicit unit meters)  and later, when we want more units, we also allow `distance` to be an object, but keep supporting simple numbers (in implicit meters)

Reply:
Hmm... that makes the schema more complex though. Also fine for me, I don't have a strong preference.

Reply:
@clausmichele @LukeWeidenwalker @soxofaan and all others:

The question basically is, how do we encode values with units when passed from users to processes.

What's the preferred way generally for openEO?
1. One parameter (e.g. `distance`) encoded as object (e.g. `{"value": 123, "unit": "m"}`)
2. Two parameters (e.g. `distance` and `unit`) endcoded as scalars (e.g. `123` and `"m"`)

Reply:
Slight preference for `One parameter (e.g. distance) encoded as object (e.g. {"value": 123, "unit": "m"})`

Reply:
I'm also for just one parameter as an object including value and unit.